src/components/inputs/EthAddressInput.tsx

```diff
@@ -13,68 +13,80 @@ export function EthAddressInput({
   placeholder = `juicebox.eth / ${constants.AddressZero}`,
   onChange,
 }: {
-  placeholder?: string
   value?: string
+  placeholder?: string
   onChange?: (value: string) => void
 }) {
-  const [input, setInput] = useState<string>(value ?? '')
+  const [input, setInput] = useState<string>()
 
   const [addressForENSName, setAddressForENSName] = useState<string>()
   const [loadingENSName, setLoadingENSName] = useState<boolean>(false)
   const [ensName, setENSName] = useState<string>()
 
-  const triggerChange = (changedValue?: string) => {
-    onChange?.(changedValue ?? input)
-  }
+  const handleENSInput = useCallback(
+    async (ens: string) => {
+      setENSName(ens)
 
-  const handleENSInput = async (ens: string) => {
-    const addressForENSName = await readProvider.resolveName(ens)
-    if (addressForENSName) {
-      setAddressForENSName(addressForENSName)
+      const addressForENSName = await readProvider.resolveName(ens)
+      if (addressForENSName) {
+        setAddressForENSName(addressForENSName)
+        onChange?.(addressForENSName)
+      }
+    },
+    [onChange],
+  )
 
-      triggerChange(addressForENSName)
-      setENSName(ens)
-    }
-  }
-
-  const handleAddressInput = async (address: string) => {
-    const ensNameForAddress = await readProvider.lookupAddress(address)
-    if (ensNameForAddress) {
-      setENSName(ensNameForAddress)
-      setAddressForENSName(address)
-      setInput(ensNameForAddress)
-      triggerChange(address)
-    }
-  }
+  const handleAddressInput = useCallback(
+    async (address: string) => {
+      onChange?.(address)
 
-  const onInputChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
-    // Clear existing
-    if (ensName || addressForENSName) {
-      setInput('')
-      setAddressForENSName(undefined)
+      const ensNameForAddress = await readProvider.lookupAddress(address)
```

Comment:
(not related to your PR) This whole logic here feels like it might be smelly. We should probably be using the `ensName` hook here, and by doing so we can probably nuke a lot of this code.

Probably out of scope though. If it's working, it's working!